### PR TITLE
Feature/ueda 8 counterparty suspension

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**
Added suspension of counterparty and agency for admin.
In requests for agent added filters for active entities. 
Maybe something I have not consider.

### CODE REVIEW

File: HappyTravel.Edo.UnitTests/Tests/AdministratorServices/CounterpartyManagementServiceTests/CounterpartySuspensionTests.cs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    public class CounterpartySuspensionTests
+    {
+        public CounterpartySuspensionTests()
+        {
+            _mockCreationHelper = new AdministratorServicesMockCreationHelper();
+        }
+
+
+        [Fact]
+        public async Task Suspension_of_not_existing_counterparty_should_fail()
```
Comment (kirillta) [inline HappyTravel.Edo.UnitTests/Tests/AdministratorServices/CounterpartyManagementServiceTests/CounterpartySuspensionTests.cs:19]: Again I consider suspension of non existing counterparty or anything should pass

File: Api/AdministratorServices/CounterpartyManagementService.cs [hunk 2]
```diff
@@ -165,6 +165,157 @@ async Task<Result> CreatePaymentAccountsForAgencies(Counterparty counterparty)
...
+            => CheckForCounterpartySuspension(counterpartyId)
+                .BindWithTransaction(_context, Suspend);
+
+
+        public Task<Result> SuspendAgency(int agencyId)
+            => CheckForAgencySuspension(agencyId)
+                .BindWithTransaction(_context, Suspend);
+
+
+        private async Task<Result<Counterparty>> CheckForCounterpartySuspension(int counterpartyId)
```
[the remark this answers is not in the document]
Comment (hasmikharutyunyan) [reply Api/AdministratorServices/CounterpartyManagementService.cs:178]: result is used in next methods, for that i can place this method as local function, maybe will be better?

File: Api/AdministratorServices/CounterpartyManagementService.cs [hunk 3]
```diff
@@ -165,6 +165,157 @@ async Task<Result> CreatePaymentAccountsForAgencies(Counterparty counterparty)
...
+            if (!agency.IsActive)
+                return Result.Failure<Agency>("Agency already suspended.");
+
+            return Result.Ok(agency);
+        }
+
+
+        private Task<Result> Suspend(Counterparty counterparty)
+        {
+            return SuspendCounterparty()
```
[the remark this answers is not in the document]
Comment (hasmikharutyunyan) [reply Api/AdministratorServices/CounterpartyManagementService.cs:204]: this method is used in SuspendCounterparty and in SuspendAgency and there exist transaction.

File: HappyTravel.Edo.UnitTests/Tests/AdministratorServices/CounterpartyManagementServiceTests/CounterpartySuspensionTests.cs (hunk 1 above)
Comment (hasmikharutyunyan) [reply HappyTravel.Edo.UnitTests/Tests/AdministratorServices/CounterpartyManagementServiceTests/CounterpartySuspensionTests.cs:19]: Why it should pass?

Comment (kirillta) [reply HappyTravel.Edo.UnitTests/Tests/AdministratorServices/CounterpartyManagementServiceTests/CounterpartySuspensionTests.cs]: No party means no action, no action means no fail. The nature of the method under the test is idempotent

File: Api/AdministratorServices/CounterpartyManagementService.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (kirillta) [reply Api/AdministratorServices/CounterpartyManagementService.cs]: In that case it's not a _Check_ but a _Get_ method

**Pull request merged**